mono/metadata/profiler.c

```diff
@@ -1214,6 +1315,13 @@ create_profiler (void)
 	prof->mempool = mono_mempool_new ();
 	return prof;
 }
+
+MonoProfiler*
+mono_profiler_create()
+{
+	return create_profiler();
```

Comment:
What is the purpose of this API?

Reply:
Create valid MonoProfiler object for new mono_profiler_set_profiler_events function added in this PR.

Reply:
I don't see why you need to have Mono create this for you. It makes no assumptions about `MonoProfiler` other than to treat it as an opaque handle I though. You can pass in any value you like?

Reply:
We thought about the opaque pointer, but decided against it as we think it would be nicer to have a valid object when inspecting the objects in a debugger and the function already existed, it just wasn't exposed.